.github/workflows/macos-microshift.yml

```diff
@@ -35,6 +35,18 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version: ${{ matrix.go }}
+      - name: check if CRC_PULL_SECRET exists
+        shell: bash
+        run: |
+          if [ "${{ secrets.CRC_PULL_SECRET }}" != '' ]; then
+             echo "the secret \"CRC_PULL_SECRET\" has not been set, please go to \"settings \> secrets \> actions\" to create it"
+             exit 1
+          fi
+      - name: Write the pull secret to json file
+        run: |
+          echo $CRC_PULL_SECRET > temp-ps.json
```

Comment:
Could this be ` echo ${{ secrets.CRC_PULL_SECRET }} > temp-ps.json` without the intermediate environment variable?

Reply:
As per https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow it should be set as env or STDIN before using it from `run`